storage/registries.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package storage
+
+import (
+	"sort"
+
+	"github.com/juju/errors"
+)
+
+// ChainedProviderRegistry is storage provider registry that combines
+// multiple storage provider registries, chaining their results. Registries
+// earlier in the chain take precedence.
+type ChainedProviderRegistry []ProviderRegistry
+
+// StorageProviderTypes implements ProviderRegistry.
+func (r ChainedProviderRegistry) StorageProviderTypes() []ProviderType {
+	var result []ProviderType
+	for _, r := range r {
+		result = append(result, r.StorageProviderTypes()...)
+	}
+	return result
+}
+
+// StorageProvider implements ProviderRegistry.
+func (r ChainedProviderRegistry) StorageProvider(t ProviderType) (Provider, error) {
+	for _, r := range r {
+		p, err := r.StorageProvider(t)
+		if err == nil {
+			return p, nil
+		}
+		if errors.IsNotFound(err) {
+			continue
+		}
+		return nil, errors.Annotatef(err, "getting storage provider %s", t)
+	}
+	return nil, errors.NotFoundf("storage provider %s", t)
+}
+
+// StaticProviderRegistry is a storage provider registry with a statically
+// defined set of providers.
+type StaticProviderRegistry struct {
+	Providers map[ProviderType]Provider
```

Comment:
I don't think it's necessary. These are never used as globals. In the one place we use it in production code (CommonStorageProviders()), we construct a new map for each call.

Added a comment.